src/coq_rules.ml

```diff
@@ -144,40 +153,56 @@ let setup_ml_deps ~scope ~loc libs =
 let coqlib_wrapper_name (s : Dune_file.Coq.t) =
   Lib_name.Local.to_string (snd s.name)
 
-let setup_rules ~sctx ~dir ~dir_contents (s : Dune_file.Coq.t) =
+let coq_modules_of_dep ~sctx lib =
+  let name = Coq_lib.name lib in
+  let dir = Coq_lib.src_dir lib in
+  let dir_contents = Dir_contents.get_without_rules sctx ~dir in
+  Dir_contents.coq_modules_of_library dir_contents ~name
 
-  let scope = SC.find_scope_by_dir sctx dir in
+let coq_theories_of_lib_deps ~lib_db =
+  List.concat_map ~f:(fun lib_dep ->
+    let loc = Dune_file.Lib_dep.get_loc lib_dep in
+    Dune_file.Lib_dep.to_lib_names lib_dep
+    |> List.map ~f:(fun name -> Result.ok_exn (Coq_lib.DB.resolve lib_db (loc, name))))
```

Comment:
I think it would be nicer if we had a `find_many` in `Coq_lib.DB` that took a list of names.

Reply:
Good idea, done!